modules/programs/borgmatic.nix

```diff
@@ -122,12 +134,26 @@ let
 
   removeNullValues = attrSet: filterAttrs (key: value: value != null) attrSet;
 
+  hmFiles = lib.mapAttrsToList (_: file: file) config.home.file;
+  hmSymlinks = (lib.filter (file: !file.recursive) hmFiles);
+  hmExcludePatterns = map (file: ''
+    ${config.home.homeDirectory}/${file.target}
+  '') hmSymlinks;
+  hmExcludeFile = pkgs.writeText "borgmatic-home-manager-exclude-symlinks.txt"
+    (lib.concatStrings hmExcludePatterns);
```

Comment:
This will not append a final newline to the file, which might be fine in this instance but will almost certainly bite us one day when we don't expect it. Lines should be newline-terminated, not -separated.

```nix
hmExcludePatterns = lib.concatMapStrings (file: "${config.home.homeDirectory}/${file.target}\n") hmSymlinks;
```

Reply:
I agree with @ncfavier: I changed my code to look like @rycee's structure but making sure all lines are terminated with a newline character.